blog/2024-07-10-uppy-4.0.md

```diff
@@ -0,0 +1,422 @@
+next to your existing [`@uppy/google-drive`](/docs/google-drive/) plugin.
+
+## UX improvements for viewing remote files
+
+When using [Dashboard](/docs/dashboard) with any of our remote sources (Google
+Drive, Facebook, etc.) you use our internal `@uppy/provider-views` plugin to
+navigate and select files. In Uppy 4.0, we are making a handful of quality of
+life improvements for users.  
+We describe the main changes in a table below.
+
+<table style="text-align:left; font-size: 15px">
+  <tbody>
+    <tr>
+      <td colspan="2">
+        <b style="font-size: 17px;">Folder states: checked, unchecked, partial</b>
+        <p style="margin-bottom: 0;">In 4.0, we introduce a new folder state - a "partially checked" folder. A folder acquires this state when some files within the folder are "checked", and some files are "unchecked".</p>
+      </td>
+    </tr>
+    <tr>
+      <th style="padding-top: 0; padding-bottom: 0;">PREVIOUSLY</th>
+      <th style="padding-top: 0; padding-bottom: 0;">NOW</th>
+    </tr>
+    <tr>
+      <td style="width: 50%;"><video src="/img/blog/4.0/partial-old.mp4" controls></video></td>
+      <td style="width: 50%;"><video src="/img/blog/4.0/partial-new.mp4" controls></video></td>
+    </tr>
+    <tr>
+      <td colspan="2">
+        <b style="font-size: 17px;">Cache</b>
+        <p style="margin-bottom: 0;">When navigating in and out of folders, you no longer have to wait for the same API call — results get cached.</p>
+      </td>
+    </tr>
+    <tr>
+      <th style="padding-top: 0; padding-bottom: 0;">PREVIOUSLY</th>
+      <th style="padding-top: 0; padding-bottom: 0;">NOW</th>
+    </tr>
+    <tr>
+      <td style="width: 50%;"><video src="/img/blog/4.0/cache-old.mp4" controls></video></td>
+      <td style="width: 50%;"><video src="/img/blog/4.0/cache-new.mp4" controls></video></td>
+    </tr>
+    <tr>
+      <td colspan="2">
+        <b style="font-size: 17px;">Restrictions</b>
+        <p style="margin-bottom: 0;">
+          Uppy supports file <a href="/docs/uppy/#restrictions">restrictions</a>, such as maximum number of files and maximum file size. In 4.0, we reworked our restrictions UI - users will get immediate feedback upon exceeding the number of selected files, and users get a chance to reenter the correct number of files after their first upload attempt.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <th style="padding-top: 0; padding-bottom: 0;">PREVIOUSLY</th>
+      <th style="padding-top: 0; padding-bottom: 0;">NOW</th>
+    </tr>
+    <tr>
+      <td style="width: 50%;"><video src="/img/blog/4.0/restrictions-old.mp4" controls></video></td>
+      <td style="width: 50%;"><video src="/img/blog/4.0/restrictions-new.mp4" controls></video></td>
+    </tr>
+  </tbody>
+</table>
+
+We’re confident this turns our interface for remote sources into the most
+advanced one out there. We’ve seen some competing libraries not even aggregating
+results beyond the first page API limit of providers.
```

Comment:
```suggestion
results beyond the first page returned by the provider API.
```

Reply:
Nice! This part of the sentence was not making sense to me, but I scratched it up to me simply being ignorant about terminology.

Made this change in my PR